build/build.go

```diff
@@ -43,7 +48,7 @@ func Build(extension core.Extension, report ResultHandler) {
 	report(Result{true, string(output), extension})
 }
 
-func Watch(extension core.Extension, integrationCtx core.IntegrationContext, report ResultHandler) {
+func Watch(extApi *api.ExtensionsApi, extension core.Extension, integrationCtx core.IntegrationContext, report ResultHandler) {
```

Comment:
I think we should separate the localization building/watching into their own processes and continue to use the report function as a way to notify the API instead of passing it along like this. The script build process is really not related to localization files at all. Maybe something like this?

```go
// In main.go
for _, extension := range cli.config.Extensions {
    go build.Watch(extension, cli.config.IntegrationContext, func(result build.Result) {
        extension := result.Extension

        if result.Success {
            extension.Development.Status = "success"
	    fmt.Println(result)
	} else {
            extension.Development.Status = "error"
	    fmt.Fprintln(os.Stderr, result)
        }
        
        api.Notify([]core.Extension{extension})
    })
		
    go build.WatchLocalization(extension, func(result build.Result) {
        extension := result.Extension

        if result.Success {
            api.Notify([]core.Extension{extension})
            fmt.Println(result)
	} else {
	    fmt.Fprintln(os.Stderr, result)
	 }
    })
}

// In build.go
func WatchLocalization(extension core.Extension, report ResultHandler) {
	directory := filepath.Join(".", extension.Development.RootDir, "locales")
	if _, err := os.Stat(directory); os.IsNotExist(err) {
		// The extension does not have a locales directory.
		return
	}

	err, extension := setLocalization(&extension)
        if err != nil {
		report(Result{false, err.Error(), extension})
	}
        report(Result{true, 'successfully built localization', extension})

	watcher, err := fsnotify.NewWatcher()
	if err != nil {
		log.Fatal(err)
	}
	defer watcher.Close()

	done := make(chan bool)
	go func() {
		for {
			select {
			case event, ok := <-watcher.Events:
				if !ok {
					return
				}

				triggers := map[string]bool{
					fsnotify.Create.String(): true,
					fsnotify.Rename.String(): true,
					fsnotify.Write.String():  true,
				}

				if triggers[event.Op.String()] {
					err, extension := setLocalization(&extension)
					if err != nil {
                                                 report(Result{false, fmt.Printf("could not resolve localization, error:", err.Error()), extension})
					}
                                          report(Result{true, 'successfully built localization', extension})
				}
			case err, ok := <-watcher.Errors:
				if !ok {
					return
				}
				log.Println("error:", err)
			}
		}
	}()

	err = watcher.Add(directory)
	log.Println("Watcher added for:", directory)
	if err != nil {
		log.Fatal(err)
	}
	<-done
}
```